### PR TITLE
gateway: build enumerate cache

### PR DESCRIPTION
**Pull request by croissanne**
Just to see what people think, I think _run_enumerate should probably just fall back on _run_enumerate_cache instead of erroring but whatever, just to tease the idea.

This should deal with the maximum timeout attached to api gateways which is 29seconds. The actual building of the cache could happen through a cloudwatch trigger, which doesn't go via the api gatway, and thus has no timeout.

### CODE REVIEW

File: src/gateway/lambda_function.py [hunk 1]
```diff
@@ -227,6 +227,50 @@ def _run_enumerate(arguments):
...
+
+    try:
+        s3c.put_object(Bucket="rpmrepo-storage", Key="data/meta/enumerate_cache.json", Body=json.dumps(results))
+    except botocore.exceptions.ClientError:
+        return _error(500)
+    return _success(json.dumps(results))
+
+
+def _enumerate_data(arguments):
+    """Compiles the data needed for the `numerate` commands
```
Comment (schuellerf) [inline src/gateway/lambda_function.py:262]:
```suggestion
    """Compiles the data needed for the `enumerate` commands
```

File: src/gateway/lambda_function.py [hunk 2]
```diff
@@ -227,6 +227,50 @@ def _run_enumerate(arguments):
         ),
     )
 
+    try:
+        obj = s3c.get_object(Bucket="rpmrepo-storage", Key="data/meta/enumerate_cache.json")
```
Comment (schuellerf) [inline src/gateway/lambda_function.py:231]: maybe check `obj.LastModified`? and report/update this somehow?

**Review by schuellerf — CHANGES_REQUESTED** [review]
Just syntactical input from my side - don't have the big picture if other options are better 🤷‍♂️

File: src/gateway/lambda_function.py (hunk 2 above)
Comment (croissanne) [reply src/gateway/lambda_function.py:231]: hmm do you mean to only update the objects that were modified?

**Pull request closed without merging**